cdap-data-fabric/src/main/java/co/cask/cdap/data2/metrics/HBaseDatasetMetricsReporter.java

```diff
@@ -109,10 +109,17 @@ private void report(Map<String, HBaseTableUtil.TableStats> datasetStat) {
         // not a user dataset
         continue;
       }
+
+      // use the first part of the dataset name, would use history if dataset name is history.objects.kv
+      if (datasetName.contains(".")) {
```

Comment:
I believe we assume everywhere that ids (i.e. names) don't have dots. If that is not enforced for dataset - please create separate bug in JIRA.

Still, it is not cool way of figuring out the dataset name. I'd assume we want to inject datasetFramework, query its datasets and compare the prefix of table name with dataset name from the framework. And use that one.

Also, comparing dataset name against hbase table name here should be extracted into DatasetSpecification - this is where we build "."-separated names